hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/security/OMGetDelegationTokenRequest.java

```diff
@@ -101,6 +115,29 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     UpdateGetDelegationTokenRequest updateGetDelegationTokenRequest =
         getOmRequest().getUpdateGetDelegationTokenRequest();
 
+    OMResponse.Builder omResponse =
+        OMResponse.newBuilder()
+            .setCmdType(OzoneManagerProtocolProtos.Type.GetDelegationToken)
+            .setStatus(OzoneManagerProtocolProtos.Status.OK)
+            .setSuccess(true);
+
+    OMClientResponse omClientResponse = null;
+
+
+    // If security is not enabled and token request is received, leader
+    // returns token null. So, check here if updatedGetDelegationTokenResponse
+    // has response set or not. If it is not set, then token is null.
+    if (!updateGetDelegationTokenRequest.getGetDelegationTokenResponse()
+        .hasResponse()) {
+      omClientResponse = new OMGetDelegationTokenResponse(null, -1L,
+          omResponse.setGetDelegationTokenResponse(
+              GetDelegationTokenResponseProto.newBuilder()).build());
+      omClientResponse.setFlushFuture(
```

Comment:
Does this mean the null token will still get written to RocksDB?

Reply:
No, it will not be written to DB. Because we have below code in OMGetDelegationTokenResponse.
```
if (ozoneTokenIdentifier != null &&
        getOMResponse().getStatus() == OzoneManagerProtocolProtos.Status.OK) {
```